src/inmanta/main.py

```diff
@@ -210,10 +211,25 @@ def project_show(client: Client, project: str) -> None:
 @click.option("--name", "-n", help="The name of the new project", required=True)
 @click.pass_obj
 def project_create(client: Client, name: str) -> None:
+    create_project(client, name)
+
+
+def create_project(client: Client, name: str) -> None:
     project = client.get_dict("create_project", "project", {"name": name})
     print_table(["Name", "Value"], [["ID", project["id"]], ["Name", project["name"]]])
 
 
+@project.command(name="generate")
+@click.option("--name", "-n", help="The name of the new project", required=True)
+@click.option("--output-dir", "-o", help="Output directory path")
+@click.pass_obj
+def project_generate(client: Client, name: str, output_dir: Optional[str]) -> None:
```

Comment:
This command fails when:
* `--output-dir` is not set.
* When `output-dir` is set, but the directory already exists.